src/stream/src/executor/source/source_backfill_executor.rs

```diff
@@ -611,11 +611,11 @@ impl<S: StateStore> SourceBackfillExecutorInner<S> {
                             }
                             _ => {}
                         }
-                        self.backfill_state_store
-                            .state_store
-                            .commit(barrier.epoch)
-                            .await?;
                     }
+                    self.backfill_state_store
+                        .state_store
+                        .commit(barrier.epoch)
+                        .await?;
```

Comment:
It seems commit must make sure previous epoch is also committed. So made this change